### DISCUSSION TITLE
page.rects get wrong results

### DISCUSSION THREAD

**Discussion by NextGuido** [jsvine/pdfplumber#863] [Ask for help with specific PDFs]
[](url)## Describe the bug

*`page.rects` get wrong results!*

## Code to reproduce the problem

```
import pdfplumber

pdf = pdfplumber.open('EVENT-5918-8ac0a93d87736d370187753417bf0df2.pdf')
page = pdf.pages[0]
page.to_image().draw_rects(page.rects).save('rects.png')
```

## PDF file

*Please attach any PDFs necessary to reproduce the problem.*
[EVENT-5918-8ac0a93d87736d370187753417bf0df2.pdf](https://github.com/jsvine/pdfplumber/files/11218651/EVENT-5918-8ac0a93d87736d370187753417bf0df2.pdf)

*If you need to redact text in a sensitive PDF, you can run it through [JoshData/pdf-redactor](https://github.com/JoshData/pdf-redactor).*

## Expected behavior

*What did you expect the result __should__ have been?*
**just like this! Not the same file to the demo!**
![rects](https://user-images.githubusercontent.com/37173179/231671605-a19013d0-424e-4419-9eb6-935df4c347fb.jpg)

## Actual behavior

*What actually happened, instead?*
![rects](https://user-images.githubusercontent.com/37173179/231670951-5623234d-4316-4fbe-9f0a-69549fb083cf.png)

## Screenshots

*If applicable, add screenshots to help explain your problem.*

## Environment

- pdfplumber version: [0.6.0]
- Python version: [3.7.13]
- OS: [Linux version 5.4.0-146-generic (buildd@lcy02-amd64-002) (gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04))]

## Additional context

*Add any other context/notes about the problem [here.*](url)

[](url)

**Comment by jsvine** [comment]
Hi @NextGuido, and thanks for your interest in this library. Unfortunately with PDFs, things that might _look_ like rectangles are sometimes drawn by the PDF in a different way. From inspecting this PDF, it seems:

- The `.rects` are just very thin rects, to the degree that they _look_ like standard lines
- The things you expect to see as rectangles are defined in an unconventional way, and appear in the parsed objects as `.curves` (which is `pdfminer.six`'s terminology for any multi-segment line that does not perfectly create a rectangle).

p.s., if you're looking to use the table-extraction methods, upgrade to `v0.9.0`, which now pulls in the `.curve` information by default.

**Reply by NextGuido** [reply]
thanks! It works with `page.curves`.
